### ISSUE TITLE
Fail to fine-tune V1.5 model with custom llama script

### ISSUE THREAD

**Issue by lijierui** [deepseek-ai/DeepSeek-Coder#144]
After fine-tuning, the model can only generate "a", "the" or "," stuff non-stop. The training loss indicating it might suffer terrible overfitting issues. Eval loss is always ~1.1
{'loss': 0.3004, 'learning_rate': 8.07967394188264e-05, 'epoch': 2.38}
{'loss': 0.2437, 'learning_rate': 7.472068675741025e-05, 'epoch': 2.44}
{'loss': 0.1805, 'learning_rate': 6.874282320543557e-05, 'epoch': 2.5}
{'loss': 0.1866, 'learning_rate': 6.288636779155621e-05, 'epoch': 2.55}
{'loss': 0.1758, 'learning_rate': 5.717406797477372e-05, 'epoch': 2.61}
{'loss': 0.14, 'learning_rate': 5.162811128947602e-05, 'epoch': 2.66}
Could you please let me know what's wrong with my fine-tuning code? I did cross-check with the deepspeed script you provided, and didn't notice any significant differences in logic.

Thank you so much!
I could fine-tune llama/mistral with the following script(some lines omitted for simplification).
The data is using openai API format where I apply `tokenizer.apply_chat_template(messages, add_generation_prompt=True, tokenize=False)` in the "text" domain, which can output decent results if used for generating rather than training.

```
model_id = "deepseek-ai/deepseek-coder-7b-instruct-v1.5"
model= AutoModelForCausalLM.from_pretrained(model_id,
                                                 config=config,
                                                 torch_dtype=torch.bfloat16,
                                                 max_memory={i: "24564MB" for i in range(torch.cuda.device_count())},
                                                 device_map='auto',
                                                 offload_folder="offload",
                                                 offload_state_dict=True,
                                                 use_safetensors=True,
                                                 use_flash_attention_2=True
                                                 )
model.config.use_cache=False
model.enable_input_require_grads()
device = torch.device("cuda")
tokenizer = AutoTokenizer.from_pretrained(model_id, padding_side="right",use_fast=True,trust_remote_code=True)

training_args = TrainingArguments(
         per_device_train_batch_size=2,
         gradient_accumulation_steps=16,
         per_gpu_eval_batch_size=1,
         gradient_checkpointing =True,
         max_grad_norm=0.3,
         num_train_epochs=3,
         learning_rate=2e-4,
         bf16=True,
         save_total_limit=4,
         save_steps=200,
         logging_steps=10,
         output_dir=output_dir,
         optim="paged_adamw_8bit",
         lr_scheduler_type="cosine",
         evaluation_strategy="steps",
         eval_steps=100,
         warmup_ratio=0.05,
         save_safetensors=False,
         #evaluation_strategy="no",
         report_to=None)

trainer = SFTTrainer(
    model=base_model,
    args=training_args,
    max_seq_length=min(seq_length, dataset_seq_len),
    train_dataset=dataset,
    eval_dataset=eval_set,
    tokenizer= tokenizer,
    dataset_text_field=dataset_text_field,
    packing=True,
    neftune_noise_alpha=5
) # #neftune_noise_alpha=5,peft_config=peft_config,
trainer.train()
trainer.save_model(output_dir)
```

**Comment by lijierui** [comment]
I'm not sure what was changed, but I reran this same script and got decent fine-tuned models. 
Closing the issue, anyone trying to fine-tune it w/o deepspeed, feel free to use this as a reference.
